app/services/request_create_service.rb

```diff
@@ -39,59 +64,44 @@ def auto_approve?
   def start_internal_approval_process(request)
     Thread.new do
       ContextService.new(request.context).with_context do
-        default_approve? ? default_approve(request) : auto_approve(request)
+        auto_approve(request)
       end
     end
   end
 
   def auto_approve(request)
     sleep_time = ENV['AUTO_APPROVAL_INTERVAL'].to_f
 
-    start_first_stage(request)
-    request.stages.each { |stage| group_auto_approve(stage, sleep_time) }
+    start_request(request)
+
+    sub_requests = request.parent? ? [request] : request.children
+
+    sub_requests.each { |req| group_auto_approve(req, sleep_time) }
   end
 
-  def group_auto_approve(stage, sleep_time)
+  def group_auto_approve(request, sleep_time)
     sleep(sleep_time)
-    ActionCreateService.new(stage.id).create(
+    ActionCreateService.new(request.id).create(
       :operation    => Action::APPROVE_OPERATION,
       :processed_by => 'system',
       :comments     => 'ok'
     )
   end
 
-  def default_approve(request)
-    request_started(request)
-    request_finished(request)
-  end
-
-  def request_started(request)
-    RequestUpdateService.new(request.id).update(
-      :state    => Request::NOTIFIED_STATE
-    )
-  end
-
-  def request_finished(request)
-    RequestUpdateService.new(request.id).update(
-      :state    => Request::FINISHED_STATE,
-      :decision => Request::APPROVED_STATUS,
-      :reason   => 'System approved'
-    )
-  end
-
-  def start_first_stage(request)
-    request_started(request)
-    ActionCreateService.new(request.stages.first.id).create(
-      :operation    => Action::NOTIFY_OPERATION,
-      :processed_by => 'system'
-    )
-  end
+  def start_request(request)
+    sub_request_ids =
```

Comment:
@bzwei can this be made into a simple method called sub_request_ids and then you can directly use it line 100